PRO-4670-Created elastic search query for cases with grants issued

kurt-uk opened the pull request:
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRO-4670


### Change description ###
Created ES search query searching for cases that have a probate document of digitalGrant and a issue date of yesterday


**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

kurt-uk committed:
PRO-4670-Created elastic search query for cases with grants issued

Created ES search query searching for cases that have a probate document of digitalGrant and a issue date of yesterday

kurt-uk committed:
Merge branch 'cnp' of github.com:hmcts/probate-back-office into PRO-4670-create-query-to-identify-cases

kurt-uk committed:
Refactored due to duplicated code, seperated the matched cases gathering logic to be called by both methods that had duplicated code

kurt-uk committed:
added test to cover catch scenario for caseMatchingService

malinovsky reviewed (commented):
@kurt-uk As I already mentioned, you shouldn't use case matching specific `uk.gov.hmcts.probate.model.ccd.raw.casematching.CaseData` class for your requirements. Please use `uk.gov.hmcts.probate.model.ccd.raw.request.CaseData` instead. It would be nice if you create new service for your code as it's not related to case matching.

kurt-uk committed:
removed the ES query from CaseMatchingService into its own service as per code review, created new object to store the cases returned that arent part of CaseMatching. Added a Case model to the request directory

kurt-uk committed:
PRO-4672 Created IronMountain file to desired specification implementing a text file builder

Created IronMountain file with all contents needed, created a text file builder that is used by the IM class to generate the desired output. Created Grantee model objects that represent how IM see data.

kurt-uk committed:
renamed package and changed line breaks due to checkstyle

kurt-uk committed:
Merge branch 'PRO-4672-build-ironmountain-file' of github.com:hmcts/probate-back-office into PRO-4673-build-excela-report

kurt-uk committed:
PRO-4673- Created excela file based on the specification

Added Excela file builder that uses the text builder to generate a file based on the specification, implemented bool checker on the text file builder to decide whether to include the last delimiter or not

kurt-uk committed:
added extra search query param to check state of case, added ES query to search for range due to new HMRC requirement

kurt-uk committed:
removed the need for passing in caseType and created it as a local constant

kurt-uk committed:
renamed the GrantMatchingService to CaseQueryService

kurt-uk committed:
removed unnessassary Case when CaseDetails can be used
